pkg/filter/filter.go

```diff
@@ -179,6 +201,24 @@ func Predicate() predicate.Predicate {
 			e.ObjectOld.GetGeneration()
 			e.ObjectOld.GetOwnerReferences()
 
+			obj := e.ObjectNew
+
+			// Required for the case when pods are deleted due to OS upgrade
```

Comment:
I am not sure if its desirable to introduce such specific behavior here. If we only check for daemonsets we could be leaving out other potential ways of deploying driver containers, such as pods and/or deployments. Same happens with the kernel affine label, we might have other updates that dont fall into that category and we would let them go through.

Try this and see if it works:
```suggestion
			if !Owned(obj) {
				return false
			}

			if e.ObjectOld.GetGeneration() == e.ObjectNew.GetGeneration() &&
				e.ObjectOld.GetResourceVersion() == e.ObjectNew.GetResourceVersion() {
				return false
			}

			if IsSpecialResource(obj) {
				log.Info(Mode+" IsSpecialResource GenerationChanged",
					"Name", obj.GetName(), "Type", reflect.TypeOf(obj).String())
				return true
			}

			log.Info(Mode+" Owned GenerationChanged",
				"Name", obj.GetName(), "Type", reflect.TypeOf(obj).String())

			if reflect.TypeOf(obj).String() == "*v1.DaemonSet" {
				err := lifecycle.UpdateDaemonSetPods(obj)
				warn.OnError(err)
			}

			return true
```